src/memory-management/review.md

```diff
@@ -65,8 +65,8 @@ fn main() {
     // String provides no guarantees about its layout, so this could lead to
     // undefined behavior.
     unsafe {
-        let (ptr, capacity, len): (usize, usize, usize) = std::mem::transmute(s1);
-        println!("ptr = {ptr:#x}, len = {len}, capacity = {capacity}");
+        let (capacity, ptr, len): (usize, usize, usize) = std::mem::transmute(s1);
+        println!("capacity = {capacity}, ptr = {ptr:#x}, len = {len}");
```

Comment:
I wonder if it would be better to use [`String::into_raw_parts](https://doc.rust-lang.org/std/string/struct.String.html#method.into_raw_parts)` here? What do you think?

Reply:
Hmm, thanks for making me aware of this, I'm just learning rust, so I didn't know about it.

It's cool, that the string library has a feature for this, and it's a cool alternative.

I wouldn't go for it for two reasons:

  - weak reason: it's still unstable feature, so you have to enable it, otherwise it's a compile error
  - stronger reason: the currently demonstrated method works with any other type that one knows the representation for, so it has a bit more of an educational value, and there is ample warning there to say that this is for debugging, not for production

Reply:
Both good points!